pip/index.py

```diff
@@ -500,7 +497,7 @@ def mkurl_pypi_url(url):
                 )
 
             raise DistributionNotFound(
-                'No distributions matching the version for %s' % req
+                'No matching distribution found for %s' % req
```

Comment:
why did you add this after my review?  :  )   
so that's 2 things to make it not simply a refactor PR anymore.